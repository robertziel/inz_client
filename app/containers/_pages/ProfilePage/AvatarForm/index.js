import React, { useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';
import FileInput from 'react-file-input-previews-base64';

import { Button, Paper } from 'components/_ui-elements';
import CroppImageModal from 'components/CroppImageModal';

import FetchedContent from 'containers/FetchedContent';
import useApiFetcher from 'containers/BackendApiConnector/fetcher';

import messages from './messages';
import Wrapper from './Wrapper';

export default function AvatarForm() {
  const [avatar, setAvatar] = useState();
  const [inputImageBase64, setInputImageBase64] = useState();

  const handleNewImageUpload = (file) => {
    setInputImageBase64(file.base64);
  };

  const fetcher = useApiFetcher();

  const fetchData = () => {
    fetcher.get({
      path: '/profile/avatar',
      afterSuccess: (result) => {
        setAvatar(result.avatar);
      },
    });
  };

  useEffect(() => fetchData(), []);

  const submitNewAvatar = (newAvatar) => {
    setInputImageBase64(undefined);
    setAvatar(newAvatar);
  };

  return (
    <Wrapper>
      <Paper>
        <FetchedContent processing={avatar === undefined || fetcher.processing}>
          <CroppImageModal
            imageBase64={inputImageBase64}
            onSubmit={submitNewAvatar}
          />
          <Button variant="contained" component="label">
            <img className="avatar" src={avatar} alt="Profile avatar" />
            <div className="set-avatar-message">
              <FormattedMessage {...messages.setAvatar} />
            </div>
            <div className="file-input">
              <FileInput
                multiple={false}
                callbackFunction={handleNewImageUpload}
                accept="image/x-png,image/gif,image/jpeg"
              />
            </div>
          </Button>
        </FetchedContent>
      </Paper>
    </Wrapper>
  );
}
