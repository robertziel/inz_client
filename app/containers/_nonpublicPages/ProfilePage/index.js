import React from 'react';
import { FormattedMessage } from 'react-intl';

import { Divider, Grid, H1 } from 'components/_ui-elements';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import AvatarForm from './AvatarForm';
import ProfileForm from './ProfileForm';
import messages from './messages';
import { useHistory } from "react-router-dom";
import { authenticationTokenSelector } from 'containers/BackendApiConnector/selectors';

function ProfilePage({ authenticationToken }) {
  const history = useHistory();

  if (authenticationToken === null) { history.push('/sign-in') };

  return (
    <Grid container>
      <Grid item xs={12}>
        <H1>
          <FormattedMessage {...messages.header} />
        </H1>
        <Divider />
      </Grid>
      <Grid item xs={12} md={4} lg={4}>
        <AvatarForm />
      </Grid>
      <Grid item xs={12} md={6} lg={5}>
        <ProfileForm />
      </Grid>
    </Grid>
  );
}

function mapStateToProps() {
  return createSelector(
    authenticationTokenSelector(),
    (authenticationToken) => ({
      authenticationToken,
    }),
  );
}

export default connect(mapStateToProps)(ProfilePage);
