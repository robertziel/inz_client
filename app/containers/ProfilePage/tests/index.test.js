import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import ProfilePage from '../index';

describe('<ProfilePage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ProfilePage />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
