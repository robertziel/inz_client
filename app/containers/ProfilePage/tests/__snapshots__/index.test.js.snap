// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage /> should render and match the snapshot 1`] = `
<h1>
  <span>
    This is the ProfilePage container!
  </span>
</h1>
`;
